lightning/src/chain/keysinterface.rs

```diff
@@ -195,9 +195,16 @@ impl Readable for SpendableOutputDescriptor {
 // TODO: We should remove Clone by instead requesting a new ChannelKeys copy when we create
 // ChannelMonitors instead of expecting to clone the one out of the Channel into the monitors.
 pub trait ChannelKeys : Send+Clone {
-	/// Gets the commitment seed for a specific commitment number
-	/// Note that the commitment number starts at (1 << 48) - 1 and counts backwards
-	fn commitment_secret(&self, idx: u64) -> [u8; 32];
+	/// Gets the per-commitment point for a specific commitment number
+	///
+	/// Note that the commitment number starts at (1 << 48) - 1 and counts backwards.
+	fn get_per_commitment_point<T: secp256k1::Signing + secp256k1::Verification>(&self, idx: u64, secp_ctx: &Secp256k1<T>) -> PublicKey;
+	/// Gets the commitment seed for a specific commitment number, thereby revoking the commitment
```

Comment:
Is this line a little misleading right now? My understanding is that, right now, this function just returns the commitment secret for a specific commitment number, it doesn't actually do any revocation in itself.

Reply:
Since this is the interface, it just documents the desired contract.  The contract is that it's an error to try to sign a revoked transaction or revoke a signed transaction.  It is true to `EnforcingChannelKeys` doesn't *yet* implement this, so I agree that ECK should have a TODO reminding users of the class that it is incomplete.  I'll add the TODO.

Reply:
I see. I'd rephrase it to: "Gets the commitment seed for a specific commitment number. This function is also responsible for revoking the commitment." Reasoning being, the word "thereby" seems implies that just by getting the commitment seed, revocation will result.

Reply:
I rephrased somewhat differently, please take a look